.github/workflows/test_service_code.yml

```diff
@@ -39,6 +39,10 @@ jobs:
         working-directory: ${{ env.workingDirectory }}
         run: 'npm run knip'
 
+      - name: Check for TS errors
+        working-directory: ${{ env.workingDirectory }}
+        run: 'npm run typecheck'
```

Comment:
I think, considering the 'weight/importance' of these checks, that it might as well be "attached" to the linting.

So to include a "postlint/prelint/pretest"-script which runs `npm run typecheck`.
That way, we don't even have to add extra steps to the CI-workflows. (and cannot 'forget' to check)

We could also run the `typecheck:dev`-script "in parallel" with the `start:dev`-script...
I'm not sure how the console-output would look like... I'll check and try it out.

Reply:
> I think, considering the 'weight/importance' of these checks, that it might as well be "attached" to the lining.

> So to include a "postlint/prelint/pretest"-script which runs npm run typecheck.
That way, we don't even have to add extra steps to the CI-workflows. (and cannot 'forget' to check)

I'm not a fan of combining scripts unless strictly necessary. If we continue this route, then one could argue that testing should be in there too. It makes it very annoying when I want to run _only_ linting. 

> We could also run the typecheck:dev-script "in parallel" with the start:dev-script...
I'm not sure how the console-output would look like... I'll check and try it out.

I tried it. It's pretty much what the SWC compiler does. I got a response from them about it but I plan on asking them to change behaviour in a separate feature request: https://github.com/nestjs/nest/issues/13760#issuecomment-2213213953